src/gui/properties/propertieswidget.cpp

```diff
@@ -517,65 +517,51 @@ void PropertiesWidget::loadUrlSeeds()
+        return fullPath;
+    }
 
-    if (m_propListModel->itemType(index) == TorrentContentModelItem::FileType)
-        openFile(index);
-    else
-        openFolder(index, false);
+    // folder type
+    const QModelIndex nameIndex {index.sibling(index.row(), TorrentContentModelItem::COL_NAME)};
+    QString folderPath {nameIndex.data().toString()};
+    for (QModelIndex modelIdx = m_propListModel->parent(nameIndex); modelIdx.isValid(); modelIdx = modelIdx.parent())
+        folderPath.prepend(modelIdx.data().toString() + QChar('/'));
+
+    const QDir saveDir {m_torrent->savePath(true)};
+    const QString fullPath {Utils::Fs::expandPath(saveDir.absoluteFilePath(folderPath))};
+    return fullPath;
 }
 
-void PropertiesWidget::openFile(const QModelIndex &index) const
+void PropertiesWidget::openDoubleClickedItem(const QModelIndex &index) const
 {
-    int i = m_propListModel->getFileIndex(index);
-    const QDir saveDir(m_torrent->savePath(true));
-    const QString filename = m_torrent->filePath(i);
-    const QString filePath = Utils::Fs::expandPath(saveDir.absoluteFilePath(filename));
-    qDebug("Trying to open file at %s", qUtf8Printable(filePath));
-    // Flush data
-    m_torrent->flushCache();
-    Utils::Gui::openPath(filePath);
+    m_torrent->flushCache();  // Flush data
+    Utils::Gui::openPath(getFullPath(index));
 }
 
-void PropertiesWidget::openFolder(const QModelIndex &index, const bool containingFolder) const
+void PropertiesWidget::openItemContainingFolder(const QModelIndex &index) const
 {
-    QString absolutePath;
-    // FOLDER
-    if (m_propListModel->itemType(index) == TorrentContentModelItem::FolderType) {
-        // Generate relative path to selected folder
-        const QModelIndex nameIndex {index.sibling(index.row(), TorrentContentModelItem::COL_NAME)};
-        QStringList pathItems {nameIndex.data().toString()};
-        QModelIndex parent = m_propListModel->parent(nameIndex);
-        while (parent.isValid()) {
-            pathItems.prepend(parent.data().toString());
-            parent = m_propListModel->parent(parent);
-        }
-        if (pathItems.isEmpty())
-            return;
+    QString path;
 
-        const QDir saveDir(m_torrent->savePath(true));
-        const QString relativePath = pathItems.join('/');
-        absolutePath = Utils::Fs::expandPath(saveDir.absoluteFilePath(relativePath));
+    if (m_propListModel->itemType(index) == TorrentContentModelItem::FolderType) {
+        // we want to open the parent folder
+        const QModelIndex parentIndex {index.parent()};
+        path = parentIndex.isValid() ? getFullPath(parentIndex) : m_torrent->savePath(true);
     }
     else {
-        const int i = m_propListModel->getFileIndex(index);
-        const QDir saveDir(m_torrent->savePath(true));
-        const QString relativePath = m_torrent->filePath(i);
-        absolutePath = Utils::Fs::expandPath(saveDir.absoluteFilePath(relativePath));
+        path = getFullPath(index);
```

Comment:
Is it correct that `path` doesn't point to parent folder in this case (but it does in the branch above)?

Reply:
The code is correct, it will open the folder and select/highlight the file.

Reply:
>The code is correct

The logic still looks confusing for me... seems it's spread across several functions.
>it will open the folder and select/highlight the file.

But what it's supposed to do in case of folder?

Reply:
Ideally for "open containing folder" action we want:
* Action on file
  Open the folder containing the file and highlight it.
* Action on folder
  Open the parent folder and highlight the folder we selected. However not every file manager support this behavior.

I found out to achieve this we need specific workaround for each OS/file managers because each one behaves differently.
PR is updated and I tested it in windows & linux.

Reply:
I am just puzzled why we use different utilities here depending on the OS. If they do the same thing, then they should be merged into a single function (with different OS dependent implementations inside). But if they actually do different things, then it would be nice to have a corresponding comment here about the fact why we use something just because the desired feature is not available on some platform.

Offtopic: `openFolderSelect` has really bad name. It's impossible to understand what's it for without digging into its code.

Reply:
>I am just puzzled why we use different utilities here depending on the OS. If they do the same thing, then they should be merged into a single function (with different OS dependent implementations inside). But if they actually do different things, then it would be nice to have a corresponding comment here about the fact why we use something just because the desired feature is not available on some platform.

I suppose you meant `MacUtils::openFiles({path});` code path?
Basically I agree about adding some comments. But as you know I don't own a mac and I'm not able to verify its effect. I'm just keeping it compatible with the old code to avoid breaking anything. To merge the code path (and add the comments), it will require someone actually to test on the mac to be sure.